src/test/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerDelegateTest.scala

```diff
@@ -172,5 +170,25 @@ class InstanceTrackerDelegateTest extends AkkaUnitTest {
       val updateValue = statusUpdate.failed.futureValue
       updateValue shouldBe cause
     }
+
+    "Spec query fails" in {
+      Given("the delegate is processing a request")
+      val f = new Fixture
+      val appId: PathId = PathId("/test")
+      val config = AllConf.withTestConfig(
+        "--instance_tracker_update_queue_size", "1",
+        "--instance_tracker_num_parallel_updates", "1")
+      lazy val delegate = new InstanceTrackerDelegate(f.metrics, f.clock, config, f.instanceTrackerProbe.ref)
+
+      And("two queued queries")
+      delegate.specInstances(appId)
+      delegate.specInstances(appId)
+
+      When("we query a third time")
+      val result = delegate.specInstances(appId)
+
+      Then("the query fails")
+      result.failed.futureValue shouldBe a[RuntimeException]
```

Comment:
Why is this not working @zen-dog and @meln1k? I checked the queue size. It is `1`. Yet, I see three `Queued query` logs and one send request to the `InstanceTrackerActor`. I start to hate streams again.